File: app/src/main/java/com/dimatest/movieapp/MainActivity.java
package com.dimatest.movieapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;

import android.os.Bundle;
import android.view.View;

import com.dimatest.movieapp.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    public NavController navController;
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        navController = Navigation.findNavController(this, R.id.main_nav_host);
        binding.backBtn.setOnClickListener(__ -> navController.navigateUp());
        navController.addOnDestinationChangedListener((controller, destination, arguments) -> {
                    binding.toolbarLayout.setVisibility((destination.getId() == R.id.splashScreenFragment) ? View.GONE : View.VISIBLE);
                    binding.backBtn.setVisibility((destination.getId() == R.id.movieListFragment) ? View.GONE : View.VISIBLE);
                }
        );
    }

    public void setTitle(String title) {
        binding.toolbarTitleTv.setText(title);
    }
}

File: app/src/main/java/com/dimatest/movieapp/MainActivity.kt
package com.dimatest.movieapp

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.navigation.NavController
import androidx.navigation.Navigation
import com.dimatest.movieapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    val navController: NavController by lazy {
        Navigation.findNavController(this, R.id.main_nav_host)
    }
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.backBtn.setOnClickListener { navController.navigateUp() }
        navController.addOnDestinationChangedListener { _, destination, _ ->
            binding.toolbarLayout.visibility = if (destination.id == R.id.splashScreenFragment) View.GONE else View.VISIBLE
            binding.backBtn.visibility = if (destination.id == R.id.movieListFragment) View.GONE else View.VISIBLE
        }
    }

    fun setTitle(title: String?) {
        binding.toolbarTitleTv.text = title
    }
}